storage/imx6ull-flashnor/flashnor.c

```diff
@@ -93,6 +100,62 @@ static void flashnor_exit(int sig)
 }
 
 
+static storage_t *flashnor_partAdd(int parentId, off_t start, size_t size)
+{
+	storage_t *dev = calloc(1, sizeof(storage_t)), *parent;
+	int err;
+	oid_t oid;
+
+	if (dev == NULL) {
+		err = -ENOMEM;
+		(void)printf("imx6ull-flashnor: failed to allocate device, err: %s\n", strerror(err));
+		return NULL;
+	}
+	parent = storage_get(parentId);
+	if (parent == NULL) {
+		err = -EINVAL;
+		(void)printf("imx6ull-flashnor: failed to find parent device, err: %s\n", strerror(err));
+		free(dev);
+		return NULL;
+	}
+
+	dev->start = start;
+	dev->size = size;
+	dev->parent = parent;
+
+	/* Only storage parameter matters in case of partition. */
+	err = flashnor_drvInit(NULL, dev);
+	if (err < 0) {
+		(void)printf("imx6ull-flashnor: failed to initialize device, err: %s\n", strerror(err));
+		free(dev);
+		return NULL;
+	}
+	optind += 2;
+
+	err = storage_add(dev, &oid);
+	if (err < 0) {
+		(void)printf("imx6ull-flashnor: failed to create partition, err: %s\n", strerror(err));
+		free(dev);
+		return NULL;
+	}
+
+	if (snprintf(flashnor_path, sizeof(flashnor_path), "/dev/%s%u", flashnor_prefix, flashnor_id++) >= sizeof(flashnor_path)) {
+		err = -ENAMETOOLONG;
```

Comment:
`storage_add` stuff is still kept by storage - after `free(dev)` it keeps dangling pointer!